[libpas] Switch to open source MTE implementation for iOS

Achierius opened the pull request:
#### d420cf4ffe22a3b0aa41be77b75422cbec87a8c2
<pre>
[libpas] Switch to open source MTE implementation for iOS
<a href="https://bugs.webkit.org/show_bug.cgi?id=300529">https://bugs.webkit.org/show_bug.cgi?id=300529</a>
<a href="https://rdar.apple.com/162125014">rdar://162125014</a>

Reviewed by Yusuke Suzuki.

Though the actual MTE enablement code is checked at runtime (and so is
not dependent on platform -- we can build it even for non-MTE devices),
for the purposes of smoothing this transition we want to first enable it
for iOS devices (which comprise a superset of those Apple SoCs which
support the feature) in case there are any platform-specific build or
performance issues with the alternative open source MTE implementation.

Canonical link: <a href="https://commits.webkit.org/301361@main">https://commits.webkit.org/301361@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/5343e82bcf7c1ec993975f165605624128777211

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/125728 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/45390 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/36141 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/132591 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/77611 "Built successfully") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/e52ddfab-c34d-432a-a368-41e67971bd13) 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/46075 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/53949 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/95785 "Passed tests") | [💥 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/63904 "An unexpected error occured. Recent messages:Printed configuration") | [✅ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/dfa358c7-bb18-49dd-804a-43283cfcfc37) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/128676 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/36837 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/112433 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/76277 "Passed tests") | | [✅ 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/6a16dca1-9f0d-4a91-b5fc-6cc2f155db0a) 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/35739 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/30619 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/76062 "Built successfully") | | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/117810 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/106615 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/30832 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/135271 "Built successfully") | | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/124235 "Built successfully and passed tests") | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/52517 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/40275 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/104251 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/52965 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/108645 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/103979 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/26480 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/49343 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/27659 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/49791 "Built successfully") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/52412 "Built successfully") | | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/157249 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/51761 "Built successfully") | | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/39360 "Passed tests") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/55111 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/53456 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

Constellation commented on Source/bmalloc/libpas/src/libpas/pas_platform.h:
```diff
@@ -215,7 +215,7 @@
 
 #ifndef PAS_USE_OPENSOURCE_MTE
-#define PAS_USE_OPENSOURCE_MTE 0
+#define PAS_USE_OPENSOURCE_MTE PAS_PLATFORM(IOS_FAMILY)
```
I think the following would be better for consistency to the other code.
```
#if PAS_PLATFORM(IOS_FAMILY)
#define PAS_USE_OPENSOURCE_MTE 1
#endif
```

Achierius committed:
[libpas] Switch to open source MTE implementation for iOS
https://bugs.webkit.org/show_bug.cgi?id=300529
rdar://162125014

Reviewed by Yusuke Suzuki.

Though the actual MTE enablement code is checked at runtime (and so is
not dependent on platform -- we can build it even for non-MTE devices),
for the purposes of smoothing this transition we want to first enable it
for iOS devices (which comprise a superset of those Apple SoCs which
support the feature) in case there are any platform-specific build or
performance issues with the alternative open source MTE implementation.

Canonical link: https://commits.webkit.org/301361@main
